browser/brave_ads/search_result_ad/search_result_ad_handler.cc

```diff
@@ -0,0 +1,293 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/brave_ads/search_result_ad/search_result_ad_handler.h"
+
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "base/containers/contains.h"
+#include "base/containers/fixed_flat_set.h"
+#include "base/containers/flat_set.h"
+#include "base/strings/string_number_conversions.h"
+#include "base/strings/string_util.h"
+#include "chrome/browser/profiles/profile.h"
+#include "content/public/browser/render_frame_host.h"
+#include "services/service_manager/public/cpp/interface_provider.h"
+#include "third_party/abseil-cpp/absl/types/optional.h"
+
+using ads::mojom::Conversion;
+using ads::mojom::ConversionPtr;
+using ads::mojom::SearchResultAd;
+using ads::mojom::SearchResultAdPtr;
+
+namespace brave_ads {
+
+namespace {
+
+constexpr char kSearchAdLinkDataType[] = "Product";
+constexpr char kContextPropertyName[] = "@context";
+constexpr char kTypePropertyName[] = "@type";
+
+constexpr char kDataPlacementId[] = "data-placement-id";
+constexpr char kDataCreativeInstanceId[] = "data-creative-instance-id";
+constexpr char kDataCreativeSetId[] = "data-creative-set-id";
+constexpr char kDataCampaignId[] = "data-campaign-id";
+constexpr char kDataAdvertiserId[] = "data-advertiser-id";
+constexpr char kDataLandingPage[] = "data-landing-page";
+constexpr char kDataHeadlineText[] = "data-headline-text";
+constexpr char kDataDescription[] = "data-description";
+constexpr char kDataRewardsValue[] = "data-rewards-value";
+constexpr char kDataConversionTypeValue[] = "data-conversion-type-value";
+constexpr char kDataConversionUrlPatternValue[] =
+    "data-conversion-url-pattern-value";
+constexpr char kDataConversionAdvertiserPublicKeyValue[] =
+    "data-conversion-advertiser-public-key-value";
+constexpr char kDataConversionObservationWindowValue[] =
+    "data-conversion-observation-window-value";
+
+constexpr auto kVettedBraveSearchHosts =
```

Comment:
we should not add another copy of this same host list. It already exists in brave_search_utils